### PR TITLE
[REEF-1057] Enable several StyleCop.CSharp.NamingRules and fix violations, part 1

### PR DESCRIPTION
**Pull request by tcNickolas**
This change enables the following NamingRules:
 * ElementMustBeginWithLowerCaseLetter
 * FieldNamesMustBeginWithLowerCaseLetter
 * VariableNamesMustNotBePrefixed
 * StaticReadonlyFieldsMustBeginWithUpperCaseLetter

This change does not resolve associated JIRA.

JIRA:
  [REEF-1057](https://issues.apache.org/jira/browse/REEF-1057)

Pull request:
  This closes #

### CODE REVIEW

**Comment by markusweimer** [comment]
LGTM, will test and merge.

**Comment by markusweimer** [comment]
I get a test failure:

```
Expected number of contexts to close (4) differs from actual number of success indicators (16)\r\nExpected: True\r\nActual:   False

   at Org.Apache.REEF.Tests.Functional.ReefFunctionalTest.ValidateSuccessForLocalRuntime(Int32 numberOfContextsToClose, Int32 numberOfTasksToFail, Int32 numberOfEvaluatorsToFail, String testFolder) in C:\src\reef\lang\cs\Org.Apache.REEF.Tests\Functional\ReefFunctionalTest.cs:line 179
   at Org.Apache.REEF.Tests.Functional.IMRU.IMRUBroadcastReduceTest.TestIMRUBroadcastReduceOnLocalRuntime() in C:\src\reef\lang\cs\Org.Apache.REEF.Tests\Functional\IMRU\IMRUBroadcastReduceTest.cs:line 38
```

This seems unrelated. Let me try again.

**Comment by tcNickolas** [comment]
This is unrelated but actually interesting. We've seen this failure first at #940, and filed REEF-1338 for it which is now resolved, so this issue should also be gone. Do you have a driver log from this failure to show to @dkm2110 ?

**Comment by markusweimer** [comment]
> Do you have a driver log from this failure to show to @dkm2110 ?

No, sorry. I was too trigger-happy with the clean...

**Comment by tcNickolas** [comment]
Since this should be unrelated to the change, you'll probably get this failure again at some point. Could you show it to @dkm2110 the next time you see it? It being IMRU, we really want to track it down.

**Comment by markusweimer** [comment]
tests pass now. Will merge. And yes, I will pay attention to the next time it happens.
